src/index.js

```diff
@@ -227,69 +227,78 @@ function EvolvClient(opts) {
    * optimization statistics.
    */
   this.confirm = function() {
-    waitFor(context, EFFECTIVE_GENOME_UPDATED, function() {
-      const remoteContext = context.remoteContext;
-      const allocations = (remoteContext.experiments || {}).allocations // undefined is a valid state, we want to know if its undefined
-      if (!store.configuration || !allocations || !allocations.length) {
-        return;
-      }
+    // eslint-disable-next-line es/no-promise
+    return new Promise(function(resolve) {
```

Comment:
Try it to make sure this works, but this should use our ponyfill `MiniPromise.createPromise()` instead.
```suggestion
    return new MiniPromise.createPromise(function(resolve) {
```

Reply:
Thanks, let me try

Reply:
Works, updated